satellite/metainfo/validation.go

```diff
@@ -121,15 +121,24 @@ func (requests *createRequests) cleanup() {
 	}
 }
 
-func (endpoint *Endpoint) validateAuth(ctx context.Context, action macaroon.Action) (_ *console.APIKeyInfo, err error) {
+func getAPIKey(ctx context.Context, header *pb.RequestHeader) (key *macaroon.APIKey, err error) {
 	defer mon.Task()(&ctx)(&err)
+	if header != nil {
+		return macaroon.ParseRawAPIKey(header.ApiKey)
+	}
+
 	keyData, ok := auth.GetAPIKey(ctx)
```

Comment:
I'm not liking that we have two ways to pass in authentication information. Wondering whether there's a nicer way to either inject the api key into the context or altternatively, get the apikey and set it on the requests.

Maybe an grpc interceptor that uses `SetHeader` if it's nil for the requests.

Reply:
well, if it helps, we won't have two once we can remove grpc completely. this is only for backwards compatibility for now (we need to support a couple uplink versions back but I'm confident we can get everyone upgraded)